### ISSUE TITLE
enhancement: Improve Headers UI in mock server

### ISSUE THREAD

**Issue by wrongsahil** [requestly/interceptor#48]
Right now, headers in mock server are added using JSON. This might create problems as headers can't be nested json.

### Checklist
- [ ] Add/Update Headers using +- button similar to headers rule
<img width="662" alt="image" src="https://github.com/requestly/interceptor/assets/16779465/11853129-5334-45dc-8b05-8c4ec3432311">

- [ ] Add validations
- [ ] Support old headers values

**Comment by Yogesh01000100** [comment]
I am interested in working on this issue. Can you assign this to me?

**Comment by wrongsahil** [comment]
Hey @Yogesh01000100 .Thanks for the interest. Assigning this to you. Let me know if you need help with anything.

**Comment by Yogesh01000100** [comment]
@wrongsahil can you tell me if I need to add an autocomplete list to these header values?

**Comment by Yogesh01000100** [comment]
> @wrongsahil can you tell me if I need to add an autocomplete list to these header values?

any specific validation rules that I need to add?

**Comment by wrongsahil** [comment]
- Yes, we can add autocomplete too. We already have a component which lists down all the common headers. [Here](https://github.com/requestly/requestly/blob/master/app/src/config/constants/sub/header-suggestions.js) are the suggestions which we are using in Headers Rule
- `isRequired` validation for both key and value should be enough for now.

**Comment by Yogesh01000100** [comment]
Yes I have included that, can you check and confirm if any changes are required?

**Comment by wrongsahil** [comment]
Awesome. Let me check the PR. Also Can you share a [Session Replay](https://requestly.io/products/session-replays) for the changes also.

**Comment by Yogesh01000100** [comment]
@wrongsahil I have added the necessary changes, can this pull request be merged now? Do inform me if any changes are required.

**Comment by Yogesh01000100** [comment]
@wrongsahil can you check the status of merging this PR?

**Comment by Ankitsharma2023** [comment]
Hi @wrongsahil , I would live to work on this issue. Can you please assign it to me.

**Comment by Kanishkrawatt** [comment]
Hey @Ankitsharma2023 thanks for the interest, Assigning this to you. Let me know if you need help with anything.

**Comment by MoksheJain** [comment]
Hey @Ankitsharma2023 , progress?

**Comment by rasesh-here** [comment]
@wrongsahil , Can you assign it to me, if its open.

**Comment by wrongsahil** [comment]
@rasesh-here Sure. Go ahead

**Comment by Kanishkrawatt** [comment]
Hey @rasesh-here  any updates on this ?

**Comment by rasesh-here** [comment]
yes, currently i'm working on the issue on requestly/requestly#3693.
i'll complete this ASAP, hope it's fine.

**Comment by rasesh-here** [comment]
to create a mock api (in local environment), i need to be signed in, but signup gives me "403 Forbidden : nginx", which blocks me from logging in.
any hints over here?

**Comment by rasesh-here** [comment]
Hey @Kanishkrawatt @wrongsahil, any pointers to help me move ahead on this task? Appreciate the help!

**Comment by Vaibhav91one** [comment]
Hi @dinex-dev @wrongsahil . Please assign this issue to me.

**Comment by Vaibhav91one** [comment]
Hi @dinex-dev. Any update on this issue?

**Comment by dinex-dev** [comment]
Hey sorry for the delay @Vaibhav91one assigning this to you

**Comment by dinex-dev** [comment]
No-Bug November is officially over, and this issue is no longer open for contributions.
Thank you for being part of our community, more opportunities coming soon!
